distribution/src/main/gateway/bin/gateway.start.bat

```diff
@@ -19,33 +19,46 @@
 @REM under the License.
 @REM
 
-@echo off
-
-if "%OS%" == "Windows_NT" SETLOCAL EnableDelayedExpansion
-rem ---------------------------------------------------------------------------
-rem Windows start script for Kaazing Gateway
-rem ---------------------------------------------------------------------------
-
-cd %~dp0
-
-
-rem A temporary variable for the location of the gateway installation,
-rem to allow determining the conf and lib subdirectories (assumed to 
-rem be siblings to this script's 'bin' directory).
-set GW_HOME=..
-
-rem You can define various Java system properties by setting the value
-rem of the GATEWAY_OPTS environment variable before calling this script.
-rem The script itself should not be changed. For example, the setting
-rem below sets the Java maximum memory to 512MB.
-if "%GATEWAY_OPTS%" == "" (
-    set GATEWAY_OPTS=-Xmx512m
-)
-
-rem Create the classpath.
-
-rem Add a directory for management support
-set JAVA_LIBRARY_PATH=%GW_HOME%\lib\sigar
-
-java %GATEWAY_OPTS% -Djava.library.path="%JAVA_LIBRARY_PATH%" -XX:+HeapDumpOnOutOfMemoryError -cp "%GW_HOME%\lib\*" org.kaazing.gateway.server.WindowsMain %*
-
+@echo off
+
+if "%OS%" == "Windows_NT" SETLOCAL EnableDelayedExpansion
+rem ---------------------------------------------------------------------------
+rem Windows start script for Kaazing Gateway
+rem ---------------------------------------------------------------------------
+
+cd %~dp0
+
+
+rem A temporary variable for the location of the gateway installation,
+rem to allow determining the conf and lib subdirectories (assumed to 
+rem be siblings to this script's 'bin' directory).
+set GW_HOME=..
+
+rem You can define various Java system properties by setting the value
+rem of the GATEWAY_OPTS environment variable before calling this script.
+rem The script itself should not be changed. For example, the setting
+rem below sets the Java maximum memory to 512MB.
+if "%GATEWAY_OPTS%" == "" (
+    set GATEWAY_OPTS=-Xmx512m
+)
+
+rem Create the classpath.
+
+rem Add a directory for management support
+set JAVA_LIBRARY_PATH=%GW_HOME%\lib\sigar
+
+rem Verify if the gateway identifier was provided (required by Agorna multiple gateway instances)
+
+set GW_ID=
+
+if NOT "%GATEWAY_OPTS%"=="%GATEWAY_OPTS:org.kaazing.gateway.management.AGRONA_ENABLED=true=%" (
```

Comment:
This performed a replace on GATEWAY_OPTS for the "org.kaazing.gateway.management.AGRONA_ENABLED" String. It then verified that the initial GATEWAY_OPTS String was different than the one with the replace performed, which indicated the fact that the substring was found in the parent String. 
Discovered that this implementation however had a bug when AGRONA_ENABLED was set to false due to the presence of the "=" character in the search String (which is the character which indicated the separation between the substring to be replace and the value to perform the replace with).
Due to this issue and due to the fact that the previous implementation is difficult to understand switched to an implementation which may be more explicit.